pkg/csi/service/vanilla/controller.go

```diff
@@ -764,11 +764,28 @@ func (c *controller) ControllerPublishVolume(ctx context.Context, req *csi.Contr
 				},
 			}
 			// Select only the backing object details.
-			queryResult, err := c.manager.VolumeManager.QueryAllVolume(ctx, queryFilter, querySelection)
-			if err != nil {
-				msg := fmt.Sprintf("QueryVolume failed for volumeID: %q. %+v", req.VolumeId, err.Error())
-				log.Error(msg)
-				return nil, status.Error(codes.Internal, msg)
+			var queryResult *cnstypes.CnsQueryResult
+			var queryAsyncNotSupported bool
+			if commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.AsyncQueryVolume) {
+				// AsyncQueryVolume feature switch is disabled
+				queryResult, err = c.manager.VolumeManager.QueryVolumeAsync(ctx, queryFilter, querySelection)
+				if err.Error() == cnsvsphere.ErrNotSupported.Error() {
```

Comment:
Will this panic if `err != nil`?

Reply:
Changed it a bit for better readability and also check for !nil